src/errors/dynamo-error.ts

```
@@ -29,3 +29,25 @@ export class ResourceNotFoundError extends DynamoError {
   }
 }
+export class ValidationError extends DynamoError {
+  constructor(message: string, originalError: Error, context: ErrorContext) {
+    super(message, originalError, context);
+    this.name = "ValidationError";
+  }
+
+  static fromDynamoError(message: string, originalError: Error, context: ErrorContext): ValidationError {
+    // Provide clearer explanations for specific error messages
+    if (message.includes("The document path provided in the update expression is invalid for update")) {
+      message +=
```

Biome (1.9.4)

[error] Reassigning a function parameter is confusing.

The parameter is declared here:

Use a local variable instead.

(lint/style/noParameterAssign)